compiler/rustc_query_system/src/query/plumbing.rs

```diff
@@ -44,6 +44,16 @@ enum QueryResult {
     Poisoned,
 }
 
+impl QueryResult {
+    /// Unwraps the query job and if poisoned will raise a [`FatalError`]
+    fn unwrap(self) -> QueryJob {
+        match self {
+            Self::Started(job) => job,
+            Self::Poisoned => FatalError.raise(),
```

Comment:
Why did you change this to `FatalError.raise()`? It should be a `panic!("something descriptive here")`, imo.

Reply:
Because within this module the QueryJob's are unwrapped four times. Two of them previously did `=> panic!()` and two did `=> FatalError.raise()`.